packages/language/src/preprocessor/instruction-interpreter.ts

```diff
@@ -89,10 +89,19 @@ function copyValue(value: Value): Value {
   }
 }
 
-function variables(context: InterpreterContext): Map<string, Variable> {
-  // Returns the correct variable map for the current scope:
-  // local (in a procedure call), otherwise global
-  return context.localVariables ?? context.variables;
+function getVariable(
+  context: InterpreterContext,
+  name: string,
+): Variable | undefined {
+  return context.localVariables?.get(name) ?? context.variables.get(name);
+}
+
+function setVariable(context: InterpreterContext, variable: Variable): void {
+  if (context.localVariables?.has(variable.name)) {
+    context.localVariables.set(variable.name, variable);
+  } else {
+    context.variables.set(variable.name, variable);
+  }
```

Comment:
This makes sense! I had this implementation too but somehow reverted it in #351.